auction-server/src/opportunity/api.rs

```diff
@@ -102,21 +107,105 @@ pub struct OpportunityParamsV1 {
     #[serde(with = "crate::serde::u256")]
     pub target_call_value: U256,
 
-    pub sell_tokens: Vec<TokenAmount>,
-    pub buy_tokens:  Vec<TokenAmount>,
+    pub sell_tokens: Vec<TokenAmountEvm>,
+    pub buy_tokens:  Vec<TokenAmountEvm>,
 }
 
 #[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug)]
 #[serde(tag = "version")]
+pub enum OpportunityParamsEvm {
+    #[serde(rename = "v1")]
+    V1(OpportunityParamsV1Evm),
+}
+
+#[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug)]
+#[serde(untagged)]
 pub enum OpportunityParams {
+    Evm(OpportunityParamsEvm),
+    Svm(OpportunityParamsSvm),
+}
+
+#[serde_as]
+#[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug)]
+pub struct TokenAmountSvm {
+    /// Token contract address
+    #[schema(example = "DUcTi3rDyS5QEmZ4BNRBejtArmDCWaPYGfN44vBJXKL5", value_type = String)]
+    #[serde_as(as = "DisplayFromStr")]
+    pub token:  Pubkey,
+    /// Token amount
+    #[schema(example = 1000)]
+    pub amount: u64,
+}
+
+// TODO descibe it
+#[serde_as]
+#[derive(Serialize, Deserialize, ToSchema, Clone, PartialEq, Debug)]
+pub struct OpportunityParamsV1KaminoSvm {
+    #[schema(example = "DUcTi3rDyS5QEmZ4BNRBejtArmDCWaPYGfN44vBJXKL5", value_type = String)]
```

Comment:
I think the only field specific to kamino is the order and these can live on a higher level struct